engine/classes/Elgg/ViewsService.php

```diff
@@ -54,11 +54,46 @@ public function __construct(\Elgg\PluginHooksService $hooks, \Elgg\Logger $logge
 		$this->hooks = $hooks;
 		$this->logger = $logger;
 	}
+	
+	/**
+	 * Takes a view name and returns the canonical name for that view.
+	 * 
+	 * @param string $alias The possibly non-canonical view name.
+	 * 
+	 * @return string The canonical view name.
+	 */
+	public function canonicalizeViewName(/*string*/ $alias) /*: string*/ {
+		if (!is_string($alias)) {
+			return false;
+		}
+		
+		$canonical = $alias;
+
+		if (strpos($canonical, "js/") === 0) {
+			$canonical = substr($canonical, 3);
+			if (!pathinfo($canonical, PATHINFO_EXTENSION)) {
+				$canonical .= ".js";
+			}
+		} else if (strpos($canonical, "css/") === 0) {
+			$canonical = substr($canonical, 4);
+			if (!pathinfo($canonical, PATHINFO_EXTENSION)) {
+				$canonical .= ".css";
+			}
+		}
+		
+		if ($alias !== $canonical) {
+			$message = "The deprecated view $alias was moved to $canonical.\n";
```

Comment:
Is the notice really necessary? Pretty much all non-core plugins are going to generate lots of them and we'll nudge them into making their plugin non-BC with 1.12.